tests/test_benchmarking/test_simulation/test_simulation.py

```diff
@@ -58,3 +59,24 @@ def test_run__works_with_each_algorithm_type(self):
                 algorithm_type=algorithm_type,
                 settings=self.settings,
             ).run(num_runs=5)
+
+    def test_run__works_with_configs(self):
+        # fixme: probably some cool way to do this with python mocks
+        tracker = {"is_called": False}
+
+        def return_val(val):
```

Comment:
wait this is dumb, I think it will always have called it just to resolve this value :'(

Reply:
Yeah, not sure what you're trying to do here. From the title, it would seem that you just want to test that the constructor works given you've passed a config? But then why do you need this extra function?

Reply:
I was trying to test that the config actually used is the correct config, essentially

So testing the actual simulation didn't use the default config